src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCachedRequest.java

```diff
@@ -0,0 +1,390 @@
+                break;
+            case EndFW.TYPE_ID:
+                final EndFW end = factory.endRO.wrap(buffer, index, index + length);
+                onEnd(end);
+                break;
+            case AbortFW.TYPE_ID:
+                final AbortFW abort = factory.abortRO.wrap(buffer, index, index + length);
+                onAbort(abort);
+                break;
+        }
+   }
+
+    void onResponseMessage(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch(msgTypeId)
+        {
+            case WindowFW.TYPE_ID:
+                final WindowFW window = factory.windowRO.wrap(buffer, index, index + length);
+                onWindow(window);
+                break;
+            case SignalFW.TYPE_ID:
+                final SignalFW signal = factory.signalRO.wrap(buffer, index, index + length);
+                onSignal(signal);
+                break;
+            case ResetFW.TYPE_ID:
+                final ResetFW reset = factory.resetRO.wrap(buffer, index, index + length);
+                onReset(reset);
+                break;
+        }
+    }
+
+    private void onBegin(
+        BeginFW begin)
+    {
+        final OctetsFW extension = begin.extension();
+        final HttpBeginExFW httpBeginFW = extension.get(factory.httpBeginExRO::wrap);
+        final ListFW<HttpHeaderFW> requestHeaders = httpBeginFW.headers();
+
+        // count all requests
+        factory.counters.requests.getAsLong();
+        factory.counters.requestsCacheable.getAsLong();
+
+        if (DEBUG)
+        {
+            System.out.printf("[%016x] ACCEPT %016x %s [received request]\n",
+                    currentTimeMillis(), acceptReplyId, getRequestURL(httpBeginFW.headers()));
+        }
+
+        DefaultCacheEntry cacheEntry = factory.defaultCache.get(requestHash);
+        final String requestAuthorizationHeader = getHeader(requestHeaders, AUTHORIZATION);
+        cacheEntry.recentAuthorizationHeader(requestAuthorizationHeader);
+
+        etagMatched = CacheUtils.isMatchByEtag(requestHeaders, cacheEntry.etag());
+        if (etagMatched)
+        {
+            if (DEBUG)
+            {
+                System.out.printf("[%016x] ACCEPT %016x %s [sent response]\n",
+                                  currentTimeMillis(), acceptReplyId, "304");
+            }
+
+            factory.writer.do304(acceptReply,
+                                 acceptRouteId,
+                                 acceptReplyId,
+                                 factory.supplyTrace.getAsLong());
+        }
+        else
+        {
+            factory.writer.doSignal(signaler,
+                                    acceptRouteId,
+                                    acceptReplyId,
+                                    factory.supplyTrace.getAsLong(),
+                                    CACHE_ENTRY_SIGNAL);
+        }
+    }
+
+    private void onData(
+        final DataFW data)
+    {
+        //NOOP
```

Comment:
Do we need to send `WINDOW` here to avoid request stall when there is a `GET` body?

Reply:
Yep.